### ISSUE TITLE
Looking for a way to have css-in-js highlights work when there are string interpolations

### ISSUE THREAD

**Issue by matthewmatician** [nvim-treesitter/nvim-treesitter#4542]
### Describe the highlighting problem

The file `queries/ecma/injections.scm` seems to do a great job of allowing me to get css syntax highlighting when put css-in-js tagged template literals in my code. Queries [like this one](https://github.com/nvim-treesitter/nvim-treesitter/blob/master/queries/ecma/injections.scm#L42-L48) work really well...

Except if I'm using any string interpolations

### Example snippet that causes the problem

```javascript
export const test = css`
  position: relative;
  color: ${colors.red20};
  z-index: 1;
`
```

### Tree-sitter parsing result

<img width="669" alt="image" src="https://user-images.githubusercontent.com/1295261/226764255-6590da26-90a5-4025-b731-c7affe431734.png">

### Example screenshot

<img width="253" alt="image" src="https://user-images.githubusercontent.com/1295261/226764398-356b0660-48bb-4f84-b82f-58cc5b15d8c9.png">

colorscheme is gruvbox:
```lua
use { "ellisonleao/gruvbox.nvim" }
```

### Expected behavior

If I take out the string interpolation, this is what it looks like:
<img width="240" alt="image" src="https://user-images.githubusercontent.com/1295261/226764718-925e4d56-0f1c-404f-97e2-680c8963fd4e.png">
The `z-index` rule is now colored correctly.

### Output of `:checkhealth nvim-treesitter`

```markdown
nvim-treesitter: require("nvim-treesitter.health").check()
========================================================================
## Installation
  - OK: `tree-sitter` found 0.20.7 (parser generator, only needed for :TSInstallFromGrammar)
  - OK: `node` found v19.8.1 (only needed for :TSInstallFromGrammar)
  - OK: `git` executable found.
  - OK: `cc` executable found. Selected from { vim.NIL, "cc", "gcc", "clang", "cl", "zig" }
    Version: Apple clang version 14.0.0 (clang-1400.0.29.202)
  - OK: Neovim was compiled with tree-sitter runtime ABI version 14 (required >=13). Parsers must be compatible with runtime ABI.

## OS Info:
{
  machine = "arm64",
  release = "22.3.0",
  sysname = "Darwin",
  version = "Darwin Kernel Version 22.3.0: Mon Jan 30 20:38:37 PST 2023; root:xnu-8792.81.3~2/RELEASE_ARM64_T6000"
}

## Parser/Features         H L F I J
  - c                   ✓ ✓ ✓ ✓ ✓
  - css                 ✓ . ✓ ✓ ✓
  - diff                ✓ . . . .
  - gitcommit           ✓ . . . ✓
  - help                ✓ . . . ✓
  - javascript          ✓ ✓ ✓ ✓ ✓
  - json                ✓ ✓ ✓ ✓ .
  - lua                 ✓ ✓ ✓ ✓ ✓
  - markdown            ✓ . ✓ . ✓
  - python              ✓ ✓ ✓ ✓ ✓
  - query               ✓ ✓ ✓ ✓ ✓
  - scss                ✓ . ✓ ✓ ✓
  - tsx                 ✓ ✓ ✓ ✓ ✓
  - typescript          ✓ ✓ ✓ ✓ ✓
  - vim                 ✓ ✓ ✓ . ✓

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang}
```

### Output of `nvim --version`

```text
NVIM v0.8.3
Build type: Release
LuaJIT 2.1.0-beta3
Compiled by brew@Ventura-arm64.local

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/opt/homebrew/Cellar/neovim/0.8.3/share/nvim"

Run :checkhealth for more info
```

### Additional context

I added the following to my vim config (to `~/.config/nvim/after/queries/ecma/injections.scm` to add support for ```css` ``` syntax. It seems to work great...

```scm
;extends

; css`...`
(call_expression
 function: ((identifier) @_name (#eq? @_name "css"))
 arguments: ((template_string) @scss
   (#offset! @scss 0 1 0 -1))
)
```

I'm wondering if it would make sense for me to try to create another injection, titled `~/.config/nvim/after/queries/scss/injections.scm`? But I'm having a hard time knowing how to write such a query.

**Comment by matthewmatician** [comment]
Never mind! Looks like this was fixed 3 days ago. I just updated to the latest version and now it's working perfectly. Thanks for all you do!

**Comment by amaanq** [comment]
When we inject CSS highlights into the string, we are directly letting the css parser run and do its job of parsing the string *as* css. If we were to split it up and apply before/after the template literal, the css parser wouldn't work and would error out still, and that'd be an expensive change anyways. I don't see a way to remedy this unless someone writes a "jscss" parser, but I find that to be overkill to fix 1 problem

**Comment by vatosarmat** [comment]
You can get `styled-components` to work, read [this](https://github.com/nvim-treesitter/nvim-treesitter/issues/1111#issuecomment-1462338074). You need to add some `treesitter` directives and fix bug in `nvim` runtime.

**Comment by amaanq** [comment]
Mmm, seems like a lot of changes in the core plugin to just get a specific use-case to work, not sure about the nvim runtime bug though

**Comment by vatosarmat** [comment]
[This](https://github.com/nvim-treesitter/nvim-treesitter/issues/1111#issuecomment-1216655480) is not a changes in the core plugin or a some sort hack, this is just a tricky configuration. And it requires [small fix](https://github.com/nvim-treesitter/nvim-treesitter/issues/1111#issuecomment-1462338074) in nvim built-in lua files to work. I have `styled-components` gracefully highlighted without unsightly `ERROR`'s in `tressitter` playground.

**Comment by amaanq** [comment]
Yes, I mean it wouldn't fit as a core module imo, the problem it solves is too niche. Of course you can make a plugin that is effectively that module to solve that issue

**Comment by vatosarmat** [comment]
Frontend development with `react` framework is not too niche, it is very mainstream. And `styled-components` is widely used thing in `react` development.

**Comment by vatosarmat** [comment]
Web development is very mainstream and cases when languages are injected into each other are not so rare in web development.

**Comment by AlanWarren** [comment]
> Yes, I mean it wouldn't fit as a core module imo, the problem it solves is too niche. Of course you can make a plugin that is effectively that module to solve that issue

I think there's likely some fundamental misunderstanding happening here. This isn't a niche use case, but a mainstream and highly popular way of composing modern front end applications with react. It's not the only way to handle styles in JS, but a lot of applications use it now days.  We have several enterprise apps using it now since Google's Material UI has made styled components their default happy-path for apps using their component library.

It would be fantastic if we could get CSS injections with embedded JS working.
